coordinator-common/src/main/java/org/apache/kafka/coordinator/common/runtime/KafkaMetricHistogram.java

```diff
@@ -125,6 +127,6 @@ public List<NamedMeasurable> stats() {
 
     @Override
     public void record(MetricConfig config, double value, long timeMs) {
-        hdrHistogram.record((long) value);
+        hdrHistogram.record(Math.min(highestTrackableValue, (long) value));
```

Comment:
moved ceiling to KafkaMetricHistogram.record() cc @dimitarndimitrov

Reply:
It may be better to do this separately.